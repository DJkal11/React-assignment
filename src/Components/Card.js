import { useEffect, useState } from "react";

function Card(props) {
  const [courses, setCourses] = useState({
    Image: "",
    title: "",
    org: "",
    url: "",
    startDate: "",
    description: ""
  })
  
    let course = props.info;

    useEffect(() => {
        setCourses({
            Image: course.media.image.large,
            title: course.name,
            org: course.org,
            url: course.blocks_url,
            startDate: course.start_display,
            description: course.short_description
        })
        

    }, [course])

  return (
<div  className="max-w-sm bg-white rounded-lg border border-gray-200 shadow-md">
    <a href={courses.url}>
        <img className="rounded-t-lg w-full h-64 object-center" src={courses.Image} alt=""/>
    </a>
    <div className="p-5">
        
            <h5 className="mb-2 text-3xl font-bold tracking-tight text-gray-900">{courses.title}</h5>
            <h4 className="mb-2 text-xl font-semibold tracking-tight text-gray-900">{courses.org}</h4>
        
        <p className="mb-3 font-normal text-gray-700">{courses.description}</p>
        <br/>
        <div className="flex flex-row justify-between gap-5">
        <a href={courses.url} className="inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 transition-colors">
            View Course
            <svg aria-hidden="true" className="ml-2 -mr-1 w-4 h-4" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fillRule="evenodd" d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z" clipRule="evenodd"></path></svg>
        </a>
        <p className="font-normal mt-2 text-gray-800">Start Date: {courses.startDate}</p>
        </div>
    </div>
    
</div>

  );
}

export default Card;