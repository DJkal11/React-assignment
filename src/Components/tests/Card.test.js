import { render, screen, cleanup } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";

import Card from "../Card"

afterEach(() => {
  cleanup()
})

test('should render Card', () => {
    const fakeData = {"blocks_url":"https://courses.edx.org/api/courses/v2/blocks/?course_id=ccx-v1%3Aadam%2BMac_APccx%2Be0d%2Bccx%403","effort":"2:00","end":null,"enrollment_start":"2015-07-21T10:00:00Z","enrollment_end":"2016-06-29T10:00:00Z","id":"ccx-v1:adam+Mac_APccx+e0d+ccx@3","media":{"banner_image":{"uri":"/asset-v1:adam+Mac_APccx+e0d+type@asset+block@images_course_image.jpg","uri_absolute":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@asset+block@images_course_image.jpg"},"course_image":{"uri":"/asset-v1:adam+Mac_APccx+e0d+type@asset+block@Davidson_EdX-68_1_.png"},"course_video":{"uri":"http://www.youtube.com/watch?v=sAnHwOL8aAs"},"image":{"raw":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@asset+block@Davidson_EdX-68_1_.png","small":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@thumbnail+block@Davidson_EdX-68_1_-png-375x200.jpg","large":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@thumbnail+block@Davidson_EdX-68_1_-png-750x400.jpg"}},"name":"ntest","number":"Mac_APccx","org":"adam","short_description":"","start":"2016-06-27T14:10:44Z","start_display":"June 27, 2016","start_type":"timestamp","pacing":"instructor","mobile_available":false,"hidden":false,"invitation_only":false,"course_id":"ccx-v1:adam+Mac_APccx+e0d+ccx@3"};
    render(<Card info={fakeData}/>)
    
})

test("Card includes correct html innerText", () => {
    const fakeData = {"blocks_url":"https://courses.edx.org/api/courses/v2/blocks/?course_id=ccx-v1%3Aadam%2BMac_APccx%2Be0d%2Bccx%403","effort":"2:00","end":null,"enrollment_start":"2015-07-21T10:00:00Z","enrollment_end":"2016-06-29T10:00:00Z","id":"ccx-v1:adam+Mac_APccx+e0d+ccx@3","media":{"banner_image":{"uri":"/asset-v1:adam+Mac_APccx+e0d+type@asset+block@images_course_image.jpg","uri_absolute":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@asset+block@images_course_image.jpg"},"course_image":{"uri":"/asset-v1:adam+Mac_APccx+e0d+type@asset+block@Davidson_EdX-68_1_.png"},"course_video":{"uri":"http://www.youtube.com/watch?v=sAnHwOL8aAs"},"image":{"raw":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@asset+block@Davidson_EdX-68_1_.png","small":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@thumbnail+block@Davidson_EdX-68_1_-png-375x200.jpg","large":"https://courses.edx.org/asset-v1:adam+Mac_APccx+e0d+type@thumbnail+block@Davidson_EdX-68_1_-png-750x400.jpg"}},"name":"ntest","number":"Mac_APccx","org":"adam","short_description":"","start":"2016-06-27T14:10:44Z","start_display":"June 27, 2016","start_type":"timestamp","pacing":"instructor","mobile_available":false,"hidden":false,"invitation_only":false,"course_id":"ccx-v1:adam+Mac_APccx+e0d+ccx@3"};
    render(<Card info={fakeData}/>)
    const headerComponent = screen.getByText("ntest");
    expect(headerComponent).toHaveTextContent("ntest");
})

test("renders Card snapshot", () => {
    const newElement = renderer.create(<Card/>).toJSON();
    expect(newElement).toMatchSnapshot();
})