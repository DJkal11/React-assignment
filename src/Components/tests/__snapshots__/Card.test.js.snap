// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card snapshot 1`] = `
<div
  className="max-w-sm bg-white rounded-lg border border-gray-200 shadow-md"
>
  <a
    href=""
  >
    <img
      alt=""
      className="rounded-t-lg w-full h-64 object-center"
      src=""
    />
  </a>
  <div
    className="p-5"
  >
    <h5
      className="mb-2 text-3xl font-bold tracking-tight text-gray-900"
    />
    <h4
      className="mb-2 text-xl font-semibold tracking-tight text-gray-900"
    />
    <p
      className="mb-3 font-normal text-gray-700"
    />
    <br />
    <div
      className="flex flex-row justify-between gap-5"
    >
      <a
        className="inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 transition-colors"
        href=""
      >
        View Course
        <svg
          aria-hidden="true"
          className="ml-2 -mr-1 w-4 h-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </a>
      <p
        className="font-normal mt-2 text-gray-800"
      >
        Start Date: 
      </p>
    </div>
  </div>
</div>
`;
